### ISSUE TITLE
Only show Cow when first visiting the site

### ISSUE THREAD

**Issue by dillchen** [hicommonwealth/commonwealth#4312]
### Description

We only want to show one loading cow once, initially on page load see current example, which shows cow before in the Body Content

https://www.loom.com/share/404f8ad37caf4f2bbea51ac827cb260a

### Engineering Requirements
- Ensure that optimizations do not cause regressions on pages that rely on chainState.

### Acceptance Criteria
- On pages which we have skeleton loader, I should only see the cow initially upon visiting the site

**Comment by dillchen** [comment]
From @mzparacha "but removing the bobber breaks things, so again we need to make sure the comps work in harmony by removing copuling for app.chain related cases or any thing that is tightly coupled"

**Comment by mzparacha** [comment]
Recap: this issue is because of chain load, when chain is loading we show the bobber.

I have been trying to come up with a minimal solution to solve this site-wide, tried deferring chain load, minimal chain load, and other things but looks like this will need some work.

I would say there are 2 solutions to this, both having their advantages 
1. We only init the chain where we need to. In most places we won't need to init the chain. This would work by having some flags in the chain adapter `isFullyInitialized` / `isPartiallyInitialized` for example. In some pages, we only need minimal chain data like `app.chain.id / network` or other data that we can optimistically set, in those cases we can optimistically set the data and when we visit any other page that requires full chain data, then we load the full chain data there. 
2. We use the current behavior (which inits the full chain) on any page we initially land. When we do this, we can set if we want to display a skeleton UI for that page. If we do then we can update the layout incrementally to handle that. Currently, we don't have the skeleton loader for all pages. 

To get going, I feel like option 2 is best here given its incremental nature and because we don't have a skeleton loader for all pages. However, i still suggest we follow the pattern in option 1 later down the road as it has many benefits of itself.

cc: @jnaviask

**Comment by jnaviask** [comment]
Option 2 is best but I don't understand what Option 1 means exactly -- when you say "init the chain", what function are you specifically referring to? Surely not `initChain`, because it is already deferred in most places. @mzparacha

**Comment by mzparacha** [comment]
> Option 2 is best for now but I don't understand what Option 1 means exactly

Currently, even when we defer chain load, we still call `initChain`.

>  when you say "init the chain", what function are you specifically referring to

By "init the chain" I mean, we call `initChain`. 

And by deferring i mean we don't call `initChain` and set some optimistic keys for `app.chain` based on what we can infer for the loaded community. 

For example: for the`/{domain}/dydx` url, we can tell that the `app.chain` object would have `id: dydx` and some other data which we can infer (i am assuming some API changes to support the optimistic data here).

It will save us some computation in the front end and possibly might untangle some app init logic, but looks like a lot of work. Again this is just a high-level idea and will need some more digging. But the overall concept is to reduce network calls that happen in `initChain`. I believe we can do some more research and ticket it up if needed.

But for now will go with Option 2, just to keep things moving.

cc: @jnaviask

**Comment by mzparacha** [comment]
Apart from the chain loading issue which we can solve using option 2 described above, there is another issue here. 

Most of the pages are lazy loaded, which causes suspense in the whole render tree, and also the active page component. This will cause the cow bobber to show up momentarily until the lazy load finishes.

**Comment by jnaviask** [comment]
@mzparacha I think we should likely plan for a larger overhaul of the initialization logic, given the context above. But let's backlog this ticket for the time being since I don't see an easy win? Unless you have a clear idea of how to proceed rn.

**Comment by mzparacha** [comment]
> I don't see an easy win? Unless you have a clear idea of how to proceed rn.

I am still thinking about the easy win (there should be a way). I will think in parallel a bit more about ths today and for now, work on https://github.com/hicommonwealth/commonwealth/issues/4975. Backloging for now.


cc: @jnaviask

**Comment by ForestMars** [comment]
Closing as stale @jnaviask @mzparacha
